qiskit/optimization/algorithms/admm_optimizer.py

```diff
@@ -156,37 +149,49 @@ def __init__(self,
         self.rho = rho_initial
 
 
+class ADMMOptimizerResult(OptimizationResult):
+    """ ADMMOptimizer Result."""
+
+    def __init__(self, x: Optional[Any] = None, fval: Optional[Any] = None,
```

Comment:
Isn't the type of `x` a `np.ndarray` and of `fval` a float? Or does ADMM return different types sometimes?

Reply:
You are right. But in terms of inheritance they are of type `Optional[Any]`. Changing the type in the constructor does not make much sense without overriding property methods from the base class.